src/chart_types/xy_chart/annotations/rect/dimensions.integration.test.ts

```diff
@@ -256,4 +256,88 @@ describe('Render rect annotation within', () => {
     expectAnnotationAtPosition(data, 'line', dataValues, rect, 1, ScaleType.Linear);
     expectAnnotationAtPosition(data, 'bar', dataValues, rect, 1, ScaleType.Linear);
   });
+
+  it('annotation with no height will take the chart dimension height', () => {
+    const store = MockStore.default({ top: 0, left: 0, width: 20, height: 200 });
+    const settings = MockGlobalSpec.settingsNoMargins();
+    const annotation = MockAnnotationSpec.rect({
+      dataValues: [{ coordinates: { x0: 2, x1: 4 } }],
+    });
+    const barWithYNoHeight = MockSeriesSpec.bar({
+      data: [
+        { x: 1, y: 0 },
+        { x: 2, y: 0 },
+        { x: 4, y: 0 },
+      ],
+    });
+    MockStore.addSpecs([settings, annotation, barWithYNoHeight], store);
+    const expected = computeAnnotationDimensionsSelector(store.getState());
+    const [resultAnnotation] = expected.get('rect_annotation_1') ?? [];
+    expect(resultAnnotation).toMatchObject({
+      rect: { height: 200 },
+    });
+  });
+  it('annotation with fit domain will render', () => {
+    const store = MockStore.default({ top: 0, left: 0, width: 20, height: 200 });
+    const settings = MockGlobalSpec.settingsNoMargins();
+    const yDomainFitted = MockGlobalSpec.axis({ domain: { fit: true } });
+    const annotation = MockAnnotationSpec.rect({
+      dataValues: [{ coordinates: { x0: 2, x1: 4 } }],
+    });
+    const bar = MockSeriesSpec.bar({
+      data: [
+        { x: 1, y: 0 },
+        { x: 2, y: 5 },
+        { x: 4, y: 10 },
+      ],
+    });
+    MockStore.addSpecs([settings, yDomainFitted, annotation, bar], store);
+    const expected = computeAnnotationDimensionsSelector(store.getState());
+    const [resultAnnotation] = expected.get('rect_annotation_1') ?? [];
+    expect(resultAnnotation).toMatchObject({
+      rect: { height: 190 },
+    });
+  });
+  it('annotation with group id should render with x0 and x1 values', () => {
+    const store = MockStore.default({ top: 0, left: 0, width: 20, height: 200 });
+    const settings = MockGlobalSpec.settingsNoMargins();
+    const annotation = MockAnnotationSpec.rect({
+      groupId: 'group1',
+      dataValues: [{ coordinates: { x0: 2, x1: 4 } }],
+    });
+    const bar = MockSeriesSpec.bar({
+      data: [
+        { x: 1, y: 0 },
+        { x: 2, y: 5 },
+        { x: 4, y: 10 },
+      ],
+    });
+    MockStore.addSpecs([settings, annotation, bar], store);
+    const expected = computeAnnotationDimensionsSelector(store.getState());
+    const [resultAnnotation] = expected.get('rect_annotation_1') ?? [];
+    expect(resultAnnotation).toMatchObject({
+      rect: { height: 200 },
+    });
+  });
```

Comment:
You could make these more explicit that the expected value is the `height` being passed.

```suggestion
  it('annotation with group id should render with x0 and x1 values', () => {
    const height = 200;
    const store = MockStore.default({ top: 0, left: 0, width: 20, height });
    const settings = MockGlobalSpec.settingsNoMargins();
    const annotation = MockAnnotationSpec.rect({
      groupId: 'group1',
      dataValues: [{ coordinates: { x0: 2, x1: 4 } }],
    });
    const bar = MockSeriesSpec.bar({
      data: [
        { x: 1, y: 0 },
        { x: 2, y: 5 },
        { x: 4, y: 10 },
      ],
    });
    MockStore.addSpecs([settings, annotation, bar], store);
    const expected = computeAnnotationDimensionsSelector(store.getState());
    const [resultAnnotation] = expected.get('rect_annotation_1') ?? [];
    expect(resultAnnotation).toMatchObject({
      rect: { height },
    });
  });
```

Reply:
Cool! Makes sense to make this explicit and clear that it's the same as the chartStore height - addressed in 63a2df1